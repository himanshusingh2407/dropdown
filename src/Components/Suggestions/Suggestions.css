.suggestions-wrapper {
    position: absolute;
    top: 30px;
    left: 10px;
    text-align: left;
    box-shadow: 0px 4px 10px 0px rgb(140 138 140);
    border-radius: 4px;
    background-color: #ffffff;
    padding-top: 8px;
}

.suggestion {
    padding: 6px 10px;
    max-width: 250px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    cursor: pointer;
}

.suggestion:hover {
    background-color: #d5e8f4;
}

.suggestion-year {
    color: #939393;
}