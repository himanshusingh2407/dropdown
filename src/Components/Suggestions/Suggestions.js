import React from 'react';
import './Suggestions.css';

function Suggestions({ data, onSelect }) {
    const suggestionList = (list) => {
        return list.map((obj) => (
            <div className="suggestion" key={obj.imdbID} onClick={() => { onSelect(obj.imdbID) }}>
                <div>{obj.Title}</div>
                <div className="suggestion-year">{obj.Year}</div>
            </div>
        ))
    }
    return (
        <div className="suggestions-wrapper">
            {suggestionList(data)}
        </div>
    )
}

export default Suggestions;