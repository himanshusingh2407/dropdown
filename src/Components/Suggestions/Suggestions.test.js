import React from 'react';
import Suggestions from './Suggestions';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const mockData = [{"Title":"Len and Company","Year":"2015","imdbID":"tt3675204","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMTYxMzQ3ZDgtNzM3OC00YzU1LTkxNTctZWI5NDFhNWJhMTAwXkEyXkFqcGdeQXVyMjUwNDgyMDA@._V1_SX300.jpg"},{"Title":"Titanic with Len Goodman","Year":"2012–","imdbID":"tt2336103","Type":"series","Poster":"N/A"},{"Title":"Len Goodman's Partners in Rhyme","Year":"2017–","imdbID":"tt7268752","Type":"series","Poster":"https://m.media-amazon.com/images/M/MV5BZjAyYTE4ZGUtNWMwNC00YjZlLThjNjYtMzZjNTAyNjM2ZjFmXkEyXkFqcGdeQXVyNjM3NzQyODU@._V1_SX300.jpg"},{"Title":"Skazka pro len","Year":"1976","imdbID":"tt6071064","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMDZlYmM4NmQtYzhkZi00YjIyLWFmZjctYTk3Y2Q0NTljZjgwXkEyXkFqcGdeQXVyNjMxODMyODU@._V1_SX300.jpg"},{"Title":"Den hen lai len","Year":"1974","imdbID":"tt0393960","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BOTVkZWUzODAtNTY0OC00ZWZlLThkODItMjk5YmZhNDI4NGI5XkEyXkFqcGdeQXVyOTIxNDQ1MTQ@._V1_SX300.jpg"},{"Title":"Len and Hugo","Year":"2006","imdbID":"tt0857387","Type":"movie","Poster":"N/A"},{"Title":"The Roots of Wolverine: A Conversation with Stan Lee and Len Wein","Year":"2009","imdbID":"tt1834891","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BZDVmYjg2MDEtZmQ5NC00MTVjLWE2NjEtYWYxNWRlMWRmMzhmXkEyXkFqcGdeQXVyMjQ5NjgwMjA@._V1_SX300.jpg"},{"Title":"Osa len' oi andres metaxy tous","Year":"2006–","imdbID":"tt1042460","Type":"series","Poster":"N/A"},{"Title":"Doodlin': Impressions of Len Lye","Year":"1987","imdbID":"tt0340674","Type":"movie","Poster":"N/A"},{"Title":"Len(n)y","Year":"2018","imdbID":"tt8389912","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BYmE3NTg5ODQtOTI0NS00YTRkLTkxMGUtOWNiZTkwODMwNjQyXkEyXkFqcGdeQXVyNTg3NDcyODk@._V1_SX300.jpg"}];
  const tree = renderer
    .create(<Suggestions data={mockData} onSelect={()=>{}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
