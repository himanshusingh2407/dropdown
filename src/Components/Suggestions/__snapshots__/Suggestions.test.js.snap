// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="suggestions-wrapper"
>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Len and Company
    </div>
    <div
      className="suggestion-year"
    >
      2015
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Titanic with Len Goodman
    </div>
    <div
      className="suggestion-year"
    >
      2012–
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Len Goodman's Partners in Rhyme
    </div>
    <div
      className="suggestion-year"
    >
      2017–
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Skazka pro len
    </div>
    <div
      className="suggestion-year"
    >
      1976
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Den hen lai len
    </div>
    <div
      className="suggestion-year"
    >
      1974
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Len and Hugo
    </div>
    <div
      className="suggestion-year"
    >
      2006
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      The Roots of Wolverine: A Conversation with Stan Lee and Len Wein
    </div>
    <div
      className="suggestion-year"
    >
      2009
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Osa len' oi andres metaxy tous
    </div>
    <div
      className="suggestion-year"
    >
      2006–
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Doodlin': Impressions of Len Lye
    </div>
    <div
      className="suggestion-year"
    >
      1987
    </div>
  </div>
  <div
    className="suggestion"
    onClick={[Function]}
  >
    <div>
      Len(n)y
    </div>
    <div
      className="suggestion-year"
    >
      2018
    </div>
  </div>
</div>
`;
